### PR TITLE
Fix rwc

### PR DESCRIPTION
**Pull request by yuit**

### CODE REVIEW

**Comment by basarat** [comment]
I did a quick search but couldn't find the answer so don't mind my intrusion / potentially dumb question, but what is `rwc`? :rose:

**Comment by DanielRosenwasser** [comment]
@basarat it stands for **R**eal **W**orld **C**ode. We have a mix of internal/external codebases that we use to take benchmarks and find potential breaks.

File: src/harness/harness.ts [hunk 1]
```diff
@@ -1623,10 +1622,10 @@ namespace Harness {
...
     }
 
-    // Regex for check if the give filePath is a library file
-    const libRegex = /lib(\.\S+)*\.d\.ts/;
-    export function isLibraryFile(filePath: string): boolean {
-        return !!libRegex.exec(Path.getFileName(filePath));
+    export function isDefaultLibraryFile(filePath: string): boolean {
+        // We need to make sure that the filePath is prefix with "lib." not just containing "lib." and end with ".d.ts"
+        const fileName = Path.getFileName(filePath);
+        return fileName.indexOf("lib.") === 0 && !!fileName.match(/\.d\.ts/);
```
Comment (DanielRosenwasser) [inline src/harness/harness.ts]: `/^lib(\..+)*\.d\.ts$/`

1. Must start with `lib`
2. Optionally has some number of non-empty pieces that start with a period.
3. Ends in `.d.ts`

Comment (DanielRosenwasser) [inline src/harness/harness.ts]: You might choose not to do that, but as a heads up, you're not properly testing that `.d.ts` occurs last. That's what the `$` is for

File: src/harness/harness.ts [hunk 2]
```diff
@@ -1623,10 +1622,10 @@ namespace Harness {
         }
     }
 
-    // Regex for check if the give filePath is a library file
-    const libRegex = /lib(\.\S+)*\.d\.ts/;
-    export function isLibraryFile(filePath: string): boolean {
-        return !!libRegex.exec(Path.getFileName(filePath));
+    export function isDefaultLibraryFile(filePath: string): boolean {
+        // We need to make sure that the filePath is prefix with "lib." not just containing "lib." and end with ".d.ts"
```
Comment (DanielRosenwasser) [inline src/harness/harness.ts]: prefixed

File: src/harness/harness.ts (hunk 1 above)
Comment (sandersn) [inline src/harness/harness.ts]: +1 why not just add a ^ and $ to the existing regex? Is it broken for other reasons?

Comment (vladima) [inline src/harness/harness.ts]: we already have `endsWith` in `program` it can be moved to utilities and reused here

**Pull request merged**